src/transformers/trainer.py

```diff
@@ -4089,6 +4097,8 @@ def push_to_hub(self, commit_message: Optional[str] = "End of training", blockin
                 model_name = Path(self.args.output_dir).name
             else:
                 model_name = self.args.hub_model_id.split("/")[-1]
+        if token is not None:
+            self.args.hub_token = token
```

Comment:
It would be better to just take the token and pass that to the `upload_folder` call, rather than modify the state

```suggestion
        token = token if token is not None else self.args.hub_token
```

Reply:
Fully agree here! :)

Reply:
Sounds good 👍 ; updated to pass token to `init_hf_repo` and `upload_folder`